### DISCUSSION TITLE
Query all rows in which either of two child relations contains property

### DISCUSSION THREAD

**Discussion by kishansripada** [supabase/supabase#7794] [Questions]
Let's say I own a bunch of malls that have stores and restaurants. **I want to query for all malls in which at least 1 store OR at least 1 restaurants is open**

There are three tables, malls, stores and restaurants. 

malls
| id  | name |
| ------------- | ------------- |
| mall 1  | Josh's Mall |
| mall 2 | Sarah's Mall  |

stores
| id  | name | isOpen | mallId (foreign key) |
| ------------- | ------------- | ------------- | ------------- |
| 1  | Express  | false |  mall 1 |
| 2 | Banana Republic  | false | mall 2  |

restaurants
| id  | name | isOpen | mallId (foreign key) |
| ------------- | ------------- | ------------- | ------------- |
| 1  | PF Changs  | true |  mall 1 |
| 2 | Panda Express  | false | mall 2  |

Each individual store and restaurant is its own row within its respective table. Every store and restaurant contains a foreign key relation to the malls table that says that it belongs to that mall. 

**I want to query for all malls in which at least 1 store OR at least 1 restaurant is in stock. Since malls only have stores and restaurants, I'm basically querying malls that have something open**

_in this case the expected result would be Josh's mall since PF Changs is open_ 

I've been working with this but it seems that you need to specify the foreign table for a .or() request.

```  const { data, error } = await supabase
      .from("malls")
      .select("name, store!inner( isOpen ), restaurants!inner( isOpen )")
      .or("store.isOpen.eq.true, restaurants.isOpen.eq.true") 
```

When I want to specify the foreign table separately, how do I include both?

```const { data, error } = await supabase
      .from("malls")
      .select("name, stores!inner( isOpen ), restaurants!inner( isOpen )")
      .or('isOpen.eq.true', { foreignTable: "stores" }) //but then where would i specify the restaurants table?
```

In psudo code, i want to do this

```const { data, error } = await supabase
      .from("malls")
      .select("name, stores!inner( isOpen ), restaurants!inner( isOpen )")
     .or(
          .filter("stores.isOpen", "eq", true)
          .filter("restaurants.isOpen", "eq", true)
        )
```

Any help would be greatly appreciated!

**Comment by GaryAustin1** [comment]
I don't believe it is possible to use a single filter on two tables, unless something has changed recently...
https://github.com/PostgREST/postgrest/discussions/2014#discussioncomment-1598919

You are probably better off writing it in sql with a Postgres function and using rpc.
Edit:  Or creating a view.

**Reply by kishansripada** [reply]
for rpc, would that require that I define the function in supabase and not in my client side code. Or is there a way to call sql directly from the client?

**Reply by GaryAustin1** [reply]
No all you can do is pass parameters from the client to the rpc.  All SQL is in the Postgres function.

**Comment by kishansripada** [comment]
does anyone have an ideas on how to write a sql query that would beform the above actions?

```
create or replace view venues
as
    select
       id
       name
       is_open
       mall_id
    from restaurants
    union all
    select
       id
       name
       is_open
       mall_id
    from stores;
```
I started off by making a view?

**Reply by steve-chavez** [reply]
Note that you can query that same view through `supabase-js` like any other table:

```sql
const { data, error } = await supabase
  .from('venues')
  .select('*')
```

https://supabase.com/docs/reference/javascript/select
